zetaclient/chains/solana/signer/signer.go

```diff
@@ -258,25 +258,36 @@ func (signer *Signer) prepareWhitelistTx(
 
 // SetGatewayAddress sets the gateway address
 func (signer *Signer) SetGatewayAddress(address string) {
+	// noop
+	if address == "" {
```

Comment:
> should we have check as other signers that same address is set?

Yes

> should we log error if empty address is set,

No e.g. if e2e is run w/o TON it will update with empty string

> address is not in solana address format?

Yes, it's already there